### DISCUSSION TITLE
Event system for actors

### DISCUSSION THREAD

**Discussion by romaintailhurat** [excaliburjs/Excalibur#1990] [Q&A]
Hi!

First, congrats for Excalibur, its a wonderful job you've done!

I'm exploring the features, but i'm currently stumbling on the event system for actors.

My goal is simple: i designed a Planet class that inherit from Actor, and i want this to listened to actor pointer event.

My implementation is [here](https://github.com/romaintailhurat/wm-proto/blob/c9b28b9d1730dc7dc5dad3d51700498a50b7e9e9/src/planet.ts).

I'm using the `onInitialize` method to enable pointer events, and then i register a function to the `pointerup` event.

But this doesn't work, no click on the visual object triggers this event.

What am i doing wrong ?

Maybe the [way i instantiate the classes](https://github.com/romaintailhurat/wm-proto/blob/c9b28b9d1730dc7dc5dad3d51700498a50b7e9e9/src/main.ts#L23) is wrong ?

**Comment by eonarheim** [comment]
@romaintailhurat Thanks for the post, I appreciate the kind remark! We've been working hard to get the next version out the door soon!

I'd expect what you're doing to work in `onInitialize()`, this feels like a sharp edge in excalibur in regards to graphics/collider behavior with pointer events. I think the issue is that even though there is a graphic, there is no collider geometry (width = 0 and height = 0) for the pointer to test against. 

![image](https://user-images.githubusercontent.com/612071/130082833-2ffd47fa-df29-4050-8fbe-c01ad5417232.png)

I'm able to reproduce this issue with the excalibur bits you're using `0.25.0-alpha.115` and the latest bits `0.25.0-alpha.178`, if I give the Actor some width and height it  appears to work as you would want.

To help with debugging the `engine.toggleDebug();` will display some very basic bounds
 
![image](https://user-images.githubusercontent.com/612071/130083435-37132609-e707-4bff-a5f1-20d39a7f36a4.png)

I'm open to improvement suggestions, perhaps a warning logged if registering a pointer event on an actor with width = 0 or height = 0?  We could perhaps add a way for pointers to test either graphic bounds or collider geometry? or both? Let me know what you think and what would have helped you sooner.

Hope this helps!

**Reply by romaintailhurat** [reply]
Thank you very much for your kind and quick reply!

I thought about a collision bounds problem but I wasn't able to grasp the mechanism by browsing the source code. 😅

I think both a warning (at compile time if it is possible with Typescript ? But runtime could be find too) and a default behaviour could be nice.

**Reply by eonarheim** [reply]
It's not released yet, but we just merged a feature to configure graphics and/or collider geometry for pointer events! #2071

**Comment by romaintailhurat** [comment]
Hi @eonarheim !

I finally chose to [define the collider shape in the constructor](https://github.com/romaintailhurat/wm-proto/blob/main/src/planet.ts#L54):

```typescript
this.body.collider.shape = Shape.Circle(this.radius);
```

It works perfectly. 👍
